snapcraft/internal/sources.py

```diff
@@ -119,8 +125,9 @@ def download(self):
 class Script(FileBase):
 
     def __init__(self, source, source_dir, source_tag=None,
-                 source_branch=None):
-        super().__init__(source, source_dir, source_tag, source_branch)
+                 source_branch=None, source_depth=None):
+        super().__init__(source, source_dir, source_tag,
+                         source_branch, source_depth)
```

Comment:
All this sounds good, but the bug mentions `source-depth` and this same modus operandi we are doing applies for `source-tag` when using local sources or tarballs

I will leave the popping for when we move the logic to the pluginhandler